### DISCUSSION TITLE
Virtualized list of items where each item is a virtualized list

### DISCUSSION THREAD

**Discussion by adventofryz** [inokawa/virtua#299] [Q&A]
I have a virtualized list of fleets which can grow up to 200.

Each fleet has about 100 vehicles. Due to excess dom elements I am trying to virtualize both the fleets and the vehicles inside each fleet.

This is what I have tried.

```javascript
import "./styles.css";
import "bootstrap/dist/css/bootstrap.min.css";
import { VList } from "virtua";
import { Card, CardHeader, Collapse } from "reactstrap";
import { useState, forwardRef } from "react";
import SimpleBar from "simplebar-react";
import "simplebar-react/dist/simplebar.min.css";

const Root = forwardRef(
  ({ children, attrs, width, height, scrolling }, ref) => {
    return (
      <SimpleBar scrollableNodeProps={{ ref }} {...attrs}>
        <div
          style={useMemo(() => {
            return {
              position: "relative",
              visibility: "hidden",
              width: width ?? "100%",
              height: height ?? "100%",
              pointerEvents: scrolling ? "none" : "auto",
            };
          }, [width, height, scrolling])}
        >
          {children}
        </div>
      </SimpleBar>
    );
  }
);

function getRandomArbitrary(min, max) {
  return Math.random() * (max - min) + min;
}

const Fleet = ({ i }) => {
  const [isOpen, setIsOpen] = useState(true);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <Card style={{ maxHeight: 200, margin: "10px 0" }} key={i}>
      <CardHeader style={{ cursor: "pointer" }} onClick={toggle}>
        Fleet {i}
      </CardHeader>
      <Collapse isOpen={isOpen} classname="card-body">
        <VList style={{ height: "auto" }}>
          {Array.from({ length: getRandomArbitrary(1, 80) }).map((_, i) => {
            return <p className="mt-2">Vehicles for {i}</p>;
          })}
        </VList>
      </Collapse>
    </Card>
  );
};

const createRows = (num) => {
  return Array.from({ length: num }).map((_, i) => {
    return <Fleet i={i} key={i} />;
  });
};

export default function App() {
  return (
    <div className="App">
      <VList style={{ height: "90vh" }} component={{ Root }}>
        {createRows(500)}
      </VList>
    </div>
  );
}
``` 

Each item is basically a card with card header as fleet name and card body which is collapsible. 

Inside card body you have virtualized list of vehicles. this list of vehicles gets scrollbar only if it exceeds the maxheight of the card otherwise it has no scrollbar and does not need to be virtualized.

**Comment by inokawa** [comment]
Virtualization in virtualized list hasn't been officially tested yet so it may work or may not work. Overriding some styles of VList may help it.

**Reply by adventofryz** [reply]
Is there any other solution to items in a virtualized list when the item is considerably larger than the height of the vlist?

Say, each card has 10 items normally, then one of the cards has 50 items. in the viewport you can only see 5 items, but because of the way virtualization works, we would be rendering this whole card in the DOM which means all 50 items in this card will be rendered in the DOM right?

**Reply by inokawa** [reply]
I'm not sure if I understand correctly, but you can decrease overscan prop to reduce rendered items outside of the viewport.
